packages/process/process_monitor.pony

```diff
@@ -565,15 +579,6 @@ actor ProcessMonitor
         _stdin_writeable = false
         _stdout_readable = false
         _stderr_readable = false
-        if _stdin_event isnt AsioEvent.none() then
```

Comment:
why not just move it right before the `_close_fd` calls in this function?

From my point of view the unsubscription does not belong in the `_close_fd` function but here inside `_close()`, unless `_close_fd(fd)` renamed/repurposed, but i would strongly vote for keeping it here.

Reply:
`_close_fd` is called from several different places besides `_close`, so `_pending_reads`, `_write_final`, `_pending_writes` all call it if the read/write operation returns EOF, and `_done_writing` also needs to unsubscribe and close the fd. Putting the unsubscribe here in `_close_fd` means that the unsubscribe is handled once in exactly the right place, right before `close`. Moving unsubscribe out of here means we need another `_unsubscribe_fd` function, or `_close_fd` could be renamed to `_unsubscribe_and_close_fd`, but conceptually unsubscribe and close should be considered to be one operation (IMO).

Reply:
Makes sense. Yes, this should be one operation.

Could you add some distilled form of your comment above to the docstring, so it is clear why unsubscribing needs to be happening here? For future reference.

Reply:
Please take a look at 05a2ea96678ab01bedee58c6ce12d1f84c0026cd and let me know what you think.

Reply:
I think this is a great docstring!